pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -89,14 +94,85 @@ func NewDynamicReconciler(rc pb.ResultsClient, lc pb.LogsClient, oc ObjectClient
 	}
 }
 
+func printGoroutines(logger *zap.SugaredLogger, o results.Object) {
+	// manual testing has confirmed you don't have to explicitly enable pprof to get goroutine dumps with
+	// stack traces; this lines up with the stack traces you receive if a panic occurs, as well as the
+	// stack trace you receive if you send a SIGQUIT and/or SIGABRT to a running go program
+	profile := pprof.Lookup("goroutine")
+	if profile == nil {
+		logger.Warnw("Leaving dynamic Reconciler only after context timeout, number of profiles found",
+			zap.String("namespace", o.GetNamespace()),
+			zap.String("kind", o.GetObjectKind().GroupVersionKind().Kind),
+			zap.String("name", o.GetName()))
+	} else {
+		err := profile.WriteTo(os.Stdout, 2)
+		if err != nil {
+			logger.Errorw("problem writing goroutine dump",
+				zap.String("error", err.Error()),
+				zap.String("namespace", o.GetNamespace()),
+				zap.String("kind", o.GetObjectKind().GroupVersionKind().Kind),
+				zap.String("name", o.GetName()))
+		}
+	}
+
+}
+
 // Reconcile handles result/record uploading for the given Run object.
 // If enabled, the object may be deleted upon successful result upload.
 func (r *Reconciler) Reconcile(ctx context.Context, o results.Object) error {
-	logger := logging.FromContext(ctx)
+	var dynamicContext context.Context
+	var dynamicCancel context.CancelFunc
+	// context with timeout does not work with the partial end to end flow that exists with unit tests;
+	// this field will alway be set for real
+	if r.cfg != nil && r.cfg.UpdateLogTimeout != nil {
+		dynamicContext, dynamicCancel = context.WithTimeout(ctx, *r.cfg.UpdateLogTimeout)
```

Comment:
```suggestion
		ctx, dynamicCancel = context.WithTimeout(ctx, *r.cfg.UpdateLogTimeout)
```

Reply:
yeah I'm not a fan of using the generic variable name for this ... I was intentional in this name change

can you elaborate @khrm on why you want the name to be `ctx` ?

Reply:
Someone modifying the function later on can easily by mistake use `ctx`. `ctx` is still available.

Reply:
I've pushed as a separate commit for now @khrm to facilitate your review  .... I can squash once we are sync'ed on the change being ready